app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -94,8 +97,28 @@ def destroyable_associations
     object.class.reflect_on_all_associations.select { |reflection| destroyable_association?(reflection) }
   end
 
+  def deletable_associations
+    object.class.reflect_on_all_associations.select { |reflection| deletable_association?(reflection) }
+  end
+
+  class BackgroundDeletionError < StandardError; end
+
   def destroyable_association?(reflection)
-    %i[destroy delete_all].include?(reflection.options[:dependent])
+    should_be_destroyed_or_deleted?(reflection, :background_destroy_associations)
+  end
+
+  def deletable_association?(reflection)
+    should_be_destroyed_or_deleted?(reflection, :background_delete_associations)
+  end
+
+  def should_be_destroyed_or_deleted?(reflection, list_name)
+    list_defined = object.respond_to?(list_name)
+
+    if !list_defined && %i[destroy delete_all].include?(reflection.options[:dependent])
```

Comment:
About my comment https://github.com/3scale/porta/pull/1372/files#r340507324
I think we should not raise an error if it is not in the list